beacon-chain/state/state-native/beacon_state_mainnet.go

```diff
@@ -45,11 +48,12 @@ type BeaconState struct {
 	latestExecutionPayloadHeader *ethpb.ExecutionPayloadHeader `ssz-gen:"true"`
 
 	lock                  sync.RWMutex
-	dirtyFields           map[types.FieldIndex]bool
-	dirtyIndices          map[types.FieldIndex][]uint64
-	stateFieldLeaves      map[types.FieldIndex]*fieldtrie.FieldTrie
-	rebuildTrie           map[types.FieldIndex]bool
+	fieldIndexPosition    map[nativetypes.FieldIndex]int
```

Comment:
Since these are compile time values, why do we need it as part of the map ? We can declare these as constants and simply use them. Or even just make it a property of the field index. `fieldIndex.RealPosition() ` There is little benefit in adding an additional map here for what would be completely static values

Reply:
I still don't understand how this will work. `RealPosition()` will be different for each state version. How do you write code like this without relying on `b`?
```
return fieldtrie.ProofFromMerkleLayers(b.merkleLayers, int(b.fieldIndexPosition[nativetypes.CurrentSyncCommittee])), nil
```